sympy/tensor/tests/test_indexed.py

```diff
@@ -353,3 +353,19 @@ def test_indexed_is_constant():
     assert not A[1+2*i, k].is_constant(i)
     assert A[1+2*i, k].is_constant(j)
     assert not A[1+2*i, k].is_constant(k)
+
+
+def test_issue_12532():
+    n = symbols("n", integer=True)
+    i = Idx('i', (1, n))
+    d = IndexedBase('d')
+    s = Sum(d[i], i)
+    s2 = s.subs(n, 5)
+    assert i.subs(Symbol("i", integer=True), 2) == 2
+    assert s2.doit() == d[1] + d[2] + d[3] + d[4] + d[5]
+
+
+def test_issue_12533():
+    d = IndexedBase('d')
+    assert d[0].subs(Symbol("d"), range(5)) == 0
```

Comment:
Yes, but it's contained in `IndexedBase`. `IndexedBase(d ==> range(5))` would give `range(5)`.